internal/crd/cache.go

```diff
@@ -6,16 +6,28 @@ import (
 	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 )
 
+type Metrics interface {
+	UpdateCrdTotal(size int)
+}
+
 type Cache struct {
-	cache *sync.Map
+	cache   *sync.Map
+	metrics Metrics
+	size    int
 }
 
-func NewCache(internalCache *sync.Map) *Cache {
-	if internalCache == nil {
-		return &Cache{cache: &sync.Map{}}
+func NewCache(cache *sync.Map, metrics Metrics) *Cache {
```

Comment:
I don't get the point of why we need to pass sync.Map as a param, the only usage in unit test is also meaningless to me.
And since we touch the refactoring of cache, I suggest we design it as a singleton.

Reply:
Yes, it's for unit testing, and therefore not meaningless. But I guess we can reach an acceptable coverage by using methods to verify. Will have a look.

Do you have an idea on how to design a singleton without using globals?